### PR TITLE
PLAT-113458: Replace straightOnlyLeave with partition config option

### PR DESCRIPTION
**Pull request by ryanjduffy**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

When navigating from a container within a `straightOnlyLeave` container, the straight-only would be ignored.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

Update the logic to:
* check upstream containers to find the first with `straightOnlyLeave` configured rather than only checkout the nearest container
* treat restricted containers (those with `enterTo` configured) as the "straight only" comparisons falling back to the target itself if it doesn't exist in any restricted containers
* limit the container search to those unique to the target to avoid comparing bounds of shared upstream containers

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

* We should refactor this logic somewhere else later. It's too much (imo) to be embedded within `spotNext`
* This lacks any automated testing atm which is definitely suboptimal but out of scope for now

### CODE REVIEW

**Comment by ryanjduffy** [comment]
@germboy - Would like your feedback on this on Monday. This "fix" is really a different implementation of the requirement that led to `straightOnlyLeave`. Let's catch up as soon as our schedules align :)

**Comment by germboy** [comment]
This change appears to look for any parent container that is configured with `straightOnlyLeave`, then use that container/dom as the reference point for moving to the next focusable element. However, it doesn't appear to actually restrict focus movement in cases where `straightOnlyLeave` is used. I think this logic will either need to be moved back to `spotNext` or there will be more to consider in the `target.js` / `navigate.js` bits of spotlight to address this.
I can provide a sample to better demonstrate my concern based on this approach, if needed.

File: packages/spotlight/src/spotlight.js [hunk 1]
```diff
@@ -335,25 +334,6 @@ const Spotlight = (function () {
...
-							(nextContainerRect.bottom < currentContainerRect.top || nextContainerRect.top > currentContainerRect.bottom)
-						) ||
-						(
-							(direction === 'up' || direction === 'down') &&
-							(nextContainerRect.right < currentContainerRect.left || nextContainerRect.left > currentContainerRect.right)
-						)
-					) {
-						return false;
-					}
-				}
```
Comment (germboy) [inline packages/spotlight/src/spotlight.js:356]: We might want to re-consolidate the multiple `if` statements above, like it was prior to the old `straightOnlyLeave` implementation
```
if (_5WayKeyHold && nextContainerIds.indexOf(currentContainerId) < 0 && !isContainer5WayHoldable(currentContainerId)) {
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,6 @@ No significant changes.
 ### Added
 
 - `core/util` function `mapAndFilterChildren` to safely iterate over React `children`
-- `spotlight` container config prop `straightOnlyLeave` to prevent navigation to oblique containers when leaving the current container
```
Comment (webOS101) [inline CHANGELOG.md:14]: :iseewhatyoudidthere:

File: packages/spotlight/src/spotlight.js (hunk 1 above)
Comment (germboy) [reply packages/spotlight/src/spotlight.js:356]: I pushed this change :)

**Pull request merged**